### PR TITLE
Google Sheets: Added sync sheet API

### PR DESCRIPTION
**Pull request by AkhileshNegi**
## Summary
 Target issue is #2530  

## Notes

- Added sync sheet API 
- Updated docs

### CODE REVIEW

File: lib/glific_web/resolvers/sheets.ex [hunk 1]
```diff
@@ -56,6 +56,22 @@ defmodule GlificWeb.Resolvers.Sheets do
...
+    with {:ok, sheet} <-
+           Repo.fetch_by(Sheet, %{id: id, organization_id: user.organization_id}),
+         {:ok, sheet} <-
+           Sheets.update_sheet(sheet, %{
+             url: sheet.url,
+             label: sheet.label,
+             organization_id: user.organization_id
+           }) do
+      {:ok, %{sheet: sheet}}
+    end
```
Comment (pankaj-ag) [inline lib/glific_web/resolvers/sheets.ex]: Did not get it. what are you trying to do here?

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: lib/glific_web/resolvers/sheets.ex (hunk 1 above)
Comment (AkhileshNegi) [reply lib/glific_web/resolvers/sheets.ex]: @pankaj-ag I'm using the sheet update function which adds `last_synced_at` in the update_sheet function and update the `sheet data` table and deletes sheet data that is not present in the sheet after recent sync

Comment (pankaj-ag) [reply lib/glific_web/resolvers/sheets.ex]: No no @AkhileshNegi that's bad practice and the code looks too coupled. 

Also, let's say if you just update the sheet label will it resync the complete sheet again? Also, you can add more columns like user roles, shortcode and updating them will perform the same operation.

So, please take out that function from the update_sheet and create a new one called sync_with_google or something. 

So whenever you will get a request to update sheet data you can update the data and call that sync function after that if needed. 

Also, last_synced_at will always be updated from the sync function and not the update one.

Comment (AkhileshNegi) [reply lib/glific_web/resolvers/sheets.ex]: @pankaj-ag made requested changes

**Review by pankaj-ag — CHANGES_REQUESTED** [review]
@AkhileshNegi let's make changes based on our offline discussion. 
Let me know if you have any question for me and we can connect quickly.

**Review by pankaj-ag — APPROVED** [review]

**Pull request merged**